### PR TITLE
[Breaking] Renamed MISTRALAI to MISTRAL everywhere

### PR DESCRIPTION
**Pull request by Sixzero**
MISTRALAI I think was deceptive, as everywhere out there it is called MISTRAL_API_KEY.

### CODE REVIEW

**Comment by Sixzero** [comment]
I doubt too many people would use Mistral models, but I hope they can also adjust this in their env vars. Although I see they need to be aware of the change. 

In their docs it is called MISTRAL_API_KEY: https://docs.mistral.ai/getting-started/quickstart/

File: src/user_preferences.jl [hunk 1]
```diff
@@ -189,9 +189,9 @@ function load_api_keys!()
     global AZURE_OPENAI_HOST
     AZURE_OPENAI_HOST = @load_preference("AZURE_OPENAI_HOST",
         default=get(ENV, "AZURE_OPENAI_HOST", ""))
-    global MISTRALAI_API_KEY
-    MISTRALAI_API_KEY = @load_preference("MISTRALAI_API_KEY",
-        default=get(ENV, "MISTRALAI_API_KEY", ""))
+    global MISTRAL_API_KEY
+    MISTRAL_API_KEY = @load_preference("MISTRAL_API_KEY",
+        default=get(ENV, "MISTRAL_API_KEY", ""))
```
Comment (cpfiffer) [inline src/user_preferences.jl:194]: Could maybe add a warning here that `MISTRALAI_API_KEY is deprecated. Here's a clumsy way of doing that.

```suggestion
    global MISTRAL_API_KEY
    if !isempty(@load_preference("MISTRALAI_API_KEY",
        default=get(ENV, "MISTRALAI_API_KEY", "")))
        @warn "The MISTRALAI_API_KEY environment variable is deprecated. Use MISTRAL_API_KEY instead"
    end
    MISTRAL_API_KEY = @load_preference("MISTRAL_API_KEY",
        default=get(ENV, "MISTRAL_API_KEY", ""))
```

**Comment by Sixzero** [comment]
Good idea!

**Comment by Sixzero** [comment]
Added the legacy support. 

On a sidenote:
Actually AISH.jl couldn't generate the solution, had to do it manually, because the file is 55k character long and most LLMs can only generate 8000 tokens as output, so for long files I guess "Instant Apply" should switch back to some legacy diffing solution.

File: CHANGELOG.md [hunk 2]
```diff
@@ -604,7 +604,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
```
Comment (svilupp) [inline CHANGELOG.md:604]:
```suggestion
### Breaking
- Changed the official ENV variable for MistralAI API from `MISTRALAI_API_KEY` to `MISTRAL_API_KEY` to be compatible with the Mistral docs.

### Added
```

**Review by svilupp — APPROVED** [review]

**Comment by svilupp** [comment]
Awesome! Thank you! I've just added the note to the changelog as a breaking change.

**Pull request merged**